server/routes/memory/routes.py

```
@@ -0,0 +1,298 @@
+
+                create_version(
+                    cursor, artifact_id, org_id, user_id, part_content,
+                    source="manual", set_current=True,
+                )
+                created_entries.append({"id": artifact_id, "title": title})
+
+            conn.commit()
+
+        logger.info(
+            f"[Memory] Uploaded {file.filename} as {category}/{base_title} "
+            f"({total_parts} part{'s' if total_parts > 1 else ''}) for org {org_id}"
```

Ruff (0.15.18)

[warning] Logging statement uses f-string

(G004)